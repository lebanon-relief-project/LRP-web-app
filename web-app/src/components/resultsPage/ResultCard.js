import React from "react";
import styled from "styled-components";

export const ResultCard = ({ title, body, id = "result card" }) => {
  return (
    <>
      <Card>
        <div id={id}>
          <h4 style={{margin: "10px"}}>
            <b>{title}</b>
          </h4>
          <p style={{margin: "10px"}}>{body}</p>
        </div>
      </Card>
    </>
  );
};

const Card = styled.div`
  box-shadow: 0 0px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  Offset: 0px, 1px
  rgba(0, 0, 0, 0.25)
  border-radius: 10px;
  padding: 6px;
  margin: 10px;
  min-height: 174px;
`;

export default ResultCard;
