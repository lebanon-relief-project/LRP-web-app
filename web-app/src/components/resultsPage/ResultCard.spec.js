import React from "react";
import { render } from "@testing-library/react";
import ResultCard from "./ResultCard";
import "jest-styled-components";
import "@testing-library/jest-dom/extend-expect";

describe("the ResultCard component", () => {
  let container;

  fit("should match the snapshot", () => {
    ({ container } = render(
      <ResultCard title="Test title" body="test body" />
    ));

    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent("Test title");
    expect(container).toHaveTextContent("test body");
  });
});
