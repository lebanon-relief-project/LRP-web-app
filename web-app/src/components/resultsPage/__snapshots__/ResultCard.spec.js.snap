// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ResultCard component should match the snapshot 1`] = `
.c0 {
  box-shadow: 0 0px 8px 0 rgba(0,0,0,0.2);
  -webkit-transition: 0.3s;
  transition: 0.3s;
  Offset: 0px,1px rgba(0,0,0,0.25) border-radius:10px;
  padding: 6px;
  margin: 10px;
  min-height: 174px;
}

<div>
  <div
    class="c0"
  >
    <div
      id="result card"
    >
      <h4
        style="margin: 10px; font-size: 20px;"
      >
        <b>
          Test title
        </b>
      </h4>
      <p
        style="margin: 10px; font-size: 16px;"
      >
        test body
      </p>
    </div>
  </div>
</div>
`;
